pkg/admission/machinedeployments_validation.go

```diff
@@ -148,8 +148,9 @@ func mutationsForMachineDeployment(md *v1alpha1.MachineDeployment, useOSM bool)
 }
 
 func ensureOSPAnnotation(md *v1alpha1.MachineDeployment, providerConfig providerconfigtypes.Config) error {
-	// Check for existing annotation
-	if _, ok := md.Annotations[osmresources.MachineDeploymentOSPAnnotation]; !ok {
+	// Check for existing annotation if it doesn't exist or if the value is empty
+	// inject the appropriate annotation.
+	if val, ok := md.Annotations[osmresources.MachineDeploymentOSPAnnotation]; !ok || val == "" {
```

Comment:
I intentionally didn't choose that because I wanted to be very explicit with the conditions for coherence. Although I can modify this if you feel the need ;)

Reply:
Zerovalue of scalar types is a handy feature, why neglect it :man_shrugging:.